functions/src/scheduler/schedule.ts

```diff
@@ -23,3 +25,28 @@ export const deletePingsAfter30days: any = functions.pubsub.schedule('00 00 * *
     });
     return Promise.all(promises);
   });
+
+export const runAllMonitors60Mins: any = functions.pubsub.schedule('every 60 mins')
+  .onRun(async () => {
+    const snapshots: QuerySnapshot = await FirebaseAdmin.firestore()
+      .collection('projects')
+      .get();
+    
+    snapshots.docs.forEach(async (doc: firestore.QueryDocumentSnapshot) => {
+      const project: ProjectModel = <ProjectModel>doc.data();
+      const promises: Promise<WriteResult>[] = [];
```

Comment:
Should this not be outside the `forEach` loop?

Reply:
We are creating list of promises for each project so we have to create promise array inside the loop. If I will defined outside the loop then it will push all

Reply:
Promise all runs in parallel, so I think creating a big list is ok?